test/e2e/suites/init/karmadactl_init_test.go

```diff
@@ -0,0 +1,567 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package init
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+const (
+	KarmadactlInitTimeOut = time.Minute * 10
+	karmadaHost           = "karmada-host"
+)
+
+// ComponentType defines the type of Kubernetes resource
+type ComponentType string
+
+const (
+	ComponentTypeDeployment  ComponentType = "deployment"
+	ComponentTypeStatefulSet ComponentType = "statefulset"
+)
+
+// ComponentConfig holds configuration for a Karmada component
+type ComponentConfig struct {
+	Name      string
+	Type      ComponentType
+	Flag      string
+	ExtraArgs []string
+	Selector  string
+}
+
+// KarmadaComponents holds all component configurations
+type KarmadaComponents struct {
+	LocalEtcd                  ComponentConfig
+	KarmadaAPIServer           ComponentConfig
+	KarmadaAggregatedAPIServer ComponentConfig
+	KubeControllerManager      ComponentConfig
+	KarmadaControllerManager   ComponentConfig
+	KarmadaScheduler           ComponentConfig
+	KarmadaWebhook             ComponentConfig
+}
+
+// NewKarmadaComponents creates a new instance of KarmadaComponents with default configurations
+func NewKarmadaComponents() *KarmadaComponents {
+	return &KarmadaComponents{
+		LocalEtcd: ComponentConfig{
+			Name:      "etcd",
+			Type:      ComponentTypeStatefulSet,
+			Flag:      "--etcd-extra-args",
+			ExtraArgs: []string{"--snapshot-count=5000", "--heartbeat-interval=100"},
+			Selector:  "app=etcd",
+		},
+		KarmadaAPIServer: ComponentConfig{
+			Name:      "karmada-apiserver",
+			Type:      ComponentTypeDeployment,
+			Flag:      "--karmada-apiserver-extra-args",
+			ExtraArgs: []string{"--tls-min-version=VersionTLS12", "--audit-log-path=-"},
+			Selector:  "app=karmada-apiserver",
+		},
+		KarmadaAggregatedAPIServer: ComponentConfig{
+			Name: "karmada-aggregated-apiserver",
+			Type: ComponentTypeDeployment,
+			Flag: "--karmada-aggregated-apiserver-extra-args",
+			ExtraArgs: []string{
+				"--tls-min-version=VersionTLS12",
+				"--audit-log-maxbackup=10",
+				"--v=2",
+				"--enable-pprof",
+			},
+			Selector: "app=karmada-aggregated-apiserver",
+		},
+		KubeControllerManager: ComponentConfig{
+			Name: "kube-controller-manager",
+			Type: ComponentTypeDeployment,
+			Flag: "--kube-controller-manager-extra-args",
```

Comment:
```suggestion
			Flag: "--karmada-kube-controller-manager-extra-args",
```

https://github.com/karmada-io/karmada/actions/runs/18774797177/job/53567427199?pr=6728#step:6:111
> error: unknown flag: --kube-controller-manager-extra-args

Reply:
Received, this has not been modified yet.